libs/langchain/langchain/retrievers/kendra.py

```diff
@@ -405,6 +440,24 @@ def _get_top_k_docs(self, result_items: Sequence[ResultItem]) -> List[Document]:
         ]
         return top_docs
 
+    def _filter_by_score_confidence(self, docs: List[Document]) -> List[Document]:
+        """
+        Filter out the records which have score confidence less than required
+        """
+        if not self.min_score_confidence:
+            return docs
+        filtered_docs = [
+            item
+            for item in docs
+            if (
+                item.metadata.get("score") is not None
+                and isinstance(item.metadata["score"], str)
```

Comment:
`get_score_attribute` is defaulting to "NOT_AVAILABLE" at the top. Is this check really required here?

Reply:
Yes. [As discussed](https://github.com/langchain-ai/langchain/pull/12920#discussion_r1384691322). In order to apply filter from `_filter_by_score_confidence`.